topics/icrc_32_sign_challenge.md

```diff
@@ -0,0 +1,110 @@
+# ICRC-32: Sign Challenge (ICRC-25 Extension)
+
+[![Status Badge](https://img.shields.io/badge/STATUS-DRAFT-ffcc00.svg)](https://github.com/orgs/dfinity/projects/31)
+
+<!-- TOC -->
+
+- [ICRC-32: Sign Challenge](#icrc-32-sign-challenge)
+  - [Summary](#summary)
+  - [Method](#method)
+  - [Request](#request-params)
+  - [Response](#request-params)
+
+## Summary
+
+The purpose of the `icrc32_sign_challenge` method is for the relying party to receive a cryptographic proof of ownership for the users identities.
+
+## Method
+
+**Name and Scope:** `icrc32_sign_challenge`
+
+**Prerequisite:** Active session with granted permission for method name/scope.
+
+## Request
+
+**`version` (`text`):** The version of the standard used. If the signer does not support the version of the request, it must send the `"VERSION_NOT_SUPPORTED"` error in response.
+
+**`challenge` (`blob`):** A challenge used for the signer to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated from a secure random source by the sender of the request.
+
+**Example RPC Request**:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc32_sign_challenge",
+  "params": {
+    "version": "1",
+    "challenge": "UjwgsORvEzp98TmB1cAIseNOoD9+GLyN/1DzJ5+jxZM="
+  }
+}
+```
+
+## Response
+
+`version` (`text`): The version of the standard used. It must match the `version` from the request.
+
+`identities`: A list of identities the user has selected to share with the relying party.
+
+- `publicKey` (`blob`): The DER-encoded public key associated with the identity, derived in accordance with one of [the signature algorithms supported by the IC](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures). The public key can be used to [derive a self-authenticating principal](https://internetcomputer.org/docs/current/references/ic-interface-spec/#principal).
```

Comment:
Relating to my comment above, I would change the interface of the method to ask for a signature by `principal`: I.e. you select an identity to sign the challenge.

This also is more sound conceptually when we allow delegations: The returned challenge signature is _not_ created by the supplied public key / principal in that case, but a delegate of the identity associated with that public key / principal.